File: app/src/main/java/ryo/myappcompany/fixingfieldsynclogger/data/AppDatabase.java
package ryo.myappcompany.fixingfieldsynclogger.data;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Report.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase instance;
    private static Context mContext;

    public abstract ReportDao reportDao();

    public static AppDatabase getInstance(Context context) {
        mContext = context;
        if (instance == null) {
            instance = Room.databaseBuilder(mContext,
                                            AppDatabase.class, "report_database")
                    .allowMainThreadQueries()
                    .build();
        }
        return instance;
    }
}

File: app/src/main/java/ryo/myappcompany/fixingfieldsynclogger/data/AppDatabase.kt
package ryo.myappcompany.fixingfieldsynclogger.data

import androidx.room.Database
import androidx.room.RoomDatabase

@Database(
    entities = [Report::class],
    version = 1
)
abstract class AppDatabase : RoomDatabase() {
    abstract fun reportDao(): ReportDao
}
